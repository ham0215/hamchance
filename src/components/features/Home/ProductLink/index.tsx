import { ReactNode } from 'react';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import { ExternalLink } from 'components/ui/ExternalLink';

type Props = {
  href: string;
  productName: string;
  name: string;
  children: ReactNode;
};

export default function ProductLayout({ href, productName, name, children }: Props) {
  return (
    <Grid item xs={12}>
      <ExternalLink href={href}>
        <Grid container spacing={4}>
          <Grid item>{children}</Grid>
          <Grid item>
            <Typography variant="body2" color="text.secondary">
              {productName}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              {name}
            </Typography>
          </Grid>
        </Grid>
      </ExternalLink>
    </Grid>
  );
}
