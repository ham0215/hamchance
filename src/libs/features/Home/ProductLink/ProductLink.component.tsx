import QrCode2Icon from '@mui/icons-material/QrCode2';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import { ExternalLink } from 'libs/ui/ExternalLink';
import { PropsWithChildren } from 'react';

type Props = {
  href: string;
  productName: string;
  name: string;
  onClickQr?: () => void;
};

export function ProductLink({ href, productName, name, onClickQr, children }: PropsWithChildren<Props>) {
  return (
    <>
      <Grid item xs={10}>
        <ExternalLink href={href}>
          <Grid container spacing={4}>
            <Grid item>{children}</Grid>
            <Grid item>
              <Typography variant="body2" color="text.secondary">
                {productName}
              </Typography>
              <Typography variant="body2" color="text.secondary">
                {name}
              </Typography>
            </Grid>
          </Grid>
        </ExternalLink>
      </Grid>
      {onClickQr && (
        <Grid item alignItems="center">
          <div onClick={onClickQr}>
            <QrCode2Icon fontSize="large" />
          </div>
        </Grid>
      )}
    </>
  );
}
