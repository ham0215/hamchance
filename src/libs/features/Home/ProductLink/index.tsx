import { ReactNode } from 'react';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import { ExternalLink } from 'libs/ui/ExternalLink';

type Props = {
  href: string;
  productName: string;
  name: string;
  children: ReactNode;
};

export function ProductLink({ href, productName, name, children }: Props) {
  return (
    <Grid item xs={12}>
      <ExternalLink href={href}>
        <Grid container spacing={4}>
          <Grid item>{children}</Grid>
          <Grid item>
            <Typography variant="body2" color="text.secondary">
              {productName}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              {name}
            </Typography>
          </Grid>
        </Grid>
      </ExternalLink>
    </Grid>
  );
}
