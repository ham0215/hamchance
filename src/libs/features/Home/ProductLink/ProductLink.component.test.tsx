import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ProductLink } from './ProductLink.component';

describe('ProductLink', () => {
  it('when set onClickQr, then QR code will be displayed', () => {
    const { asFragment } = render(
      <ProductLink href="https://hamchance.com" productName="hamchance" name="@ham" onClickQr={jest.fn()} />
    );

    expect(screen.getByText('hamchance')).toBeInTheDocument();
    expect(screen.getByTestId('QrCode2Icon')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('when set onClickQr, then QR code will be not displayed', () => {
    const { asFragment } = render(<ProductLink href="https://hamchance.com" productName="hamchance" name="@ham" />);
    expect(screen.getByText('hamchance')).toBeInTheDocument();
    expect(screen.queryByTestId('QrCode2Icon')).not.toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('when click QR code, call onClickQr', async () => {
    const onClickQr = jest.fn();
    const user = userEvent.setup();
    render(<ProductLink href="https://hamchance.com" productName="hamchance" name="@ham" onClickQr={onClickQr} />);

    act(() => {
      user.click(screen.getByTestId('QrCode2Icon'));
    });

    await waitFor(() => {
      expect(onClickQr).toHaveBeenCalledTimes(1);
    });
  });
});
