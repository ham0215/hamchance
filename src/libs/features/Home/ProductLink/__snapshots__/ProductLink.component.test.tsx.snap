// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductLink when set onClickQr, then QR code will be displayed 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-19qjfsi-MuiTypography-root-MuiLink-root"
      href="https://hamchance.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="MuiStack-root css-1xcvc1q-MuiStack-root"
      >
        <div />
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-jrj0v0-MuiTypography-root"
          >
            hamchance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-jrj0v0-MuiTypography-root"
          >
            @ham
          </p>
        </div>
      </div>
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="QrCode2Icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 21h-2v-2h2v2zm-2-7h-2v5h2v-5zm8-2h-2v4h2v-4zm-2-2h-2v2h2v-2zM7 12H5v2h2v-2zm-2-2H3v2h2v-2zm7-5h2V3h-2v2zm-7.5-.5v3h3v-3h-3zM9 9H3V3h6v6zm-4.5 7.5v3h3v-3h-3zM9 21H3v-6h6v6zm7.5-16.5v3h3v-3h-3zM21 9h-6V3h6v6zm-2 10v-3h-4v2h2v3h4v-2h-2zm-2-7h-4v2h4v-2zm-4-2H7v2h2v2h2v-2h2v-2zm1-1V7h-2V5h-2v4h4zM6.75 5.25h-1.5v1.5h1.5v-1.5zm0 12h-1.5v1.5h1.5v-1.5zm12-12h-1.5v1.5h1.5v-1.5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ProductLink when set onClickQr, then QR code will be not displayed 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-19qjfsi-MuiTypography-root-MuiLink-root"
      href="https://hamchance.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="MuiStack-root css-1xcvc1q-MuiStack-root"
      >
        <div />
        <div>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-jrj0v0-MuiTypography-root"
          >
            hamchance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-jrj0v0-MuiTypography-root"
          >
            @ham
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
